.github/workflows/build_deploy.yml

```diff
@@ -0,0 +1,81 @@
+---
+name: Build and Deploy test
+on:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: $${{ github.ref }}-$${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions: {}
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+jobs:
+  initialise:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+  build_test:
+    needs: initialise
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        test_build_conf:
+          - { commit_msg: 'Initial commit', keywords: 'SKIP-BUILD SKIP-CI' }
+          - { commit_msg: 'SKIP-BUILD Initial commit', keywords: 'SKIP-BUILD SKIP-CI' }
+          - { commit_msg: 'SKIP-CI Initial commit', keywords: 'SKIP-BUILD SKIP-CI' }
+          - { commit_msg: '[SKIP-CI] Initial commit', keywords: '[SKIP-BUILD] [SKIP-CI]' }
```

Comment:
nice that, most test cases are added here. Should also add, compatibility of different runner.